### PR TITLE
Avoid importing `.json` files

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

I'm starting to check what we need to do in order to migrate to native ES modules, and this is one of the problems I found: native ESM cannot import `.json` files by default.

This PR replaces imports to `package.json` files with a compile-time macro, which is replaced with the corresponding `package.json` entry.

- This doesn't cause problems when releasing a new version, because we compile _after_ bumping the version in `package.json`.
- This has the advantage of slightly reducing the `@babel/standalone` size, since it doesn't need to include the full `package.json` contents.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/12759"></a>

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
@kaicataldo Do you know if there is an eslint rule to say something like "disallow `*.json` imports"? I tried looking both at the core rules and at `eslint-plugin-import`, but I couldn't find anything.

**Comment by existentialism** [comment]
@nicolo-ribaudo can we use https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/extensions.md?

**Comment by nicolo-ribaudo** [comment]
That doesn't really prevent us from importing json files, it just ensures that we explicitly write the `.json` extension.

**Comment by nicolo-ribaudo** [comment]
Uh I missed some json files

**Comment by existentialism** [comment]
While not foolproof, can't we use it to at least disallow importing something with the json extension?

**Comment by nicolo-ribaudo** [comment]
Actually, probably it's better to do the opposite: if we _require_ the `.json` extension, we'll avoid accidentally importing json files.

**Comment by nicolo-ribaudo** [comment]
The new `.json` imports are all removed by https://github.com/babel/babel/pull/12583

File: babel.config.js [hunk 1]
```diff
@@ -319,3 +322,49 @@ function pluginToggleBabel8Breaking({ types: t }, { breaking }) {
...
+        if (!path.get("object").isIdentifier({ name: fnName })) return;
+
+        if (path.node.computed) {
+          throw path.buildCodeFrameError(
+            `"${fnName}" does not support computed properties.`
+          );
+        }
+        const field = path.node.property.name;
+
+        // TODO: When dropping old Node.js versions, use require.resolve
```
Comment (JLHwung) [inline babel.config.js:348]: Do we still need to support old versions since 1) we are building on latest node 2) it is a compile-time macro?

**Review by JLHwung — APPROVED** [review]
What do you think extract it to a dedicated plugin in yarn workspace. No need to publish now but maybe we can decide it later. Unlike the `BABEL_8_BREAKING` flag remover, it falls to an optimizer plugin that will stay for a long time.

File: babel.config.js (hunk 1 above)
Comment (nicolo-ribaudo) [reply babel.config.js:348]: "compile time" means also "on Node.js 6 inside babel-jest"

Comment (JLHwung) [reply babel.config.js:348]: Oh I thought we are replacing `packageJson.field` on the build step.

Comment (nicolo-ribaudo) [reply babel.config.js:348]: Yes, but we also run Babel in jest (that's the reason why we cannot use `babel.config.mjs`)

**Comment by nicolo-ribaudo** [comment]
> What do you think extract it to a dedicated plugin in yarn workspace. No need to publish now but maybe we can decide it later. Unlike the BABEL_8_BREAKING flag remover, it falls to an optimizer plugin that will stay for a long time.

We would still need to run Babel inside the yarn plugin to modify the code (unless we want to hack it with regexps) 😕

**Review by existentialism — APPROVED** [review]

**Pull request merged**